### PR TITLE
SAK-31358 Simplify add participants to two screens

### PR DESCRIPTION
**Pull request by davidpbauer**
This simplifies the add participants workflow down to two screens. The first
screen simply has a text box to enter multiple users that you would like to add
to the site. This also combined the official and non-official entries into one box.
The second page allows the user to set the role for each participant, send email
or not, and mark the users as active or inactive. There is no confirmation screen.

### CODE REVIEW

File: site-manage/site-manage-participant-helper/src/java/org/sakaiproject/site/tool/helper/participant/impl/SiteAddParticipantHandler.java [hunk 1]
```diff
@@ -938,10 +871,10 @@ else if (eidsForAllMatches.length() == 0)
 			}
 		} // officialAccounts
 
-		if (nonOfficialAccounts != null) {
-			String[] nonOfficialAccountArray = nonOfficialAccounts.split("\r\n");
-			for (i = 0; i < nonOfficialAccountArray.length; i++) {
-				String currentNonOfficialAccount = nonOfficialAccountArray[i];
+		if (!nonOfficialAccountArray.isEmpty() && getAllowNonOfficialAccount().equalsIgnoreCase("true")) {
```
Comment (bjones86) [inline site-manage/site-manage-participant-helper/src/java/org/sakaiproject/site/tool/helper/participant/impl/SiteAddParticipantHandler.java]: Flip the operands here for null safety:

```
... && "true".equalsIgnoreCase( getAllowNonOfficialAccount() )
```

File: site-manage/site-manage-participant-helper/src/java/org/sakaiproject/site/tool/helper/participant/rsf/AddProducer.java [hunk 2]
```diff
@@ -90,79 +90,27 @@ public void fillComponents(UIContainer tofill, ViewParameters arg1, ComponentChe
...
 		if (pickerAction != null && !"".equals(pickerAction))
 		{
 			UIOutput.make(participantForm, "officialAccountPickerLabel", handler.getServerConfigurationString("officialAccountPickerLabel"));
 			UIOutput.make(participantForm, "officialAccountPickerAction", pickerAction);
 		}
-    	
-		// non official participant
-    	String allowAddNonOfficialParticipant = handler.getAllowNonOfficialAccount();
+
+		String allowAddNonOfficialParticipant = handler.getAllowNonOfficialAccount();
```
Comment (bjones86) [inline site-manage/site-manage-participant-helper/src/java/org/sakaiproject/site/tool/helper/participant/rsf/AddProducer.java]: Whitespace inconsistency here

File: site-manage/site-manage-participant-helper/src/java/org/sakaiproject/site/tool/helper/participant/rsf/AddProducer.java [hunk 3]
```diff
@@ -90,79 +90,27 @@ public void fillComponents(UIContainer tofill, ViewParameters arg1, ComponentChe
...
-        }
-        
-    	UICommand.make(participantForm, "continue", messageLocator.getMessage("gen.continue"), "#{siteAddParticipantHandler.processGetParticipant}");
+		// all participant
+		UIInput.make(participantForm, "accountParticipant", "#{siteAddParticipantHandler.accountParticipant}", handler.accountParticipant);
+		UIOutput.make(participantForm, "accountSectionTitle", messageLocator.getMessage("accountSectionTitle"));
+		UIOutput.make(participantForm, "accountName", messageLocator.getMessage("accountName"));
+		UIOutput.make(participantForm, "accountLabel", messageLocator.getMessage("accountLabel"));
+
+		UICommand.make(participantForm, "continue", messageLocator.getMessage("gen.continue"), "#{siteAddParticipantHandler.processGetParticipant}");
```
Comment (bjones86) [inline site-manage/site-manage-participant-helper/src/java/org/sakaiproject/site/tool/helper/participant/rsf/AddProducer.java]: Whitespace above

File: site-manage/site-manage-participant-helper/src/java/org/sakaiproject/site/tool/helper/participant/rsf/DifferentRoleProducer.java [hunk 4]
```diff
@@ -78,11 +75,27 @@ public void setUserDirectoryService(UserDirectoryService userDirectoryService)
...
-    	UIBranchContainer content = UIBranchContainer.make(tofill, "content:");
-    	
+		String locationId = developerHelperService.getCurrentLocationId();
+		String siteTitle = "";
+		try {
+			Site s = siteService.getSite(locationId);
+			siteTitle = s.getTitle();
+		} catch (IdUnusedException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```
Comment (bjones86) [inline site-manage/site-manage-participant-helper/src/java/org/sakaiproject/site/tool/helper/participant/rsf/DifferentRoleProducer.java]: Don't print the stacktrace, log it

File: site-manage/site-manage-participant-helper/src/java/org/sakaiproject/site/tool/helper/participant/rsf/DifferentRoleProducer.java [hunk 5]
```diff
@@ -108,17 +144,21 @@ public void fillComponents(UIContainer tofill, ViewParameters arg1, ComponentChe
...
         	// default to userEid
-        	String userName = userEId;
+        	String lastName = "";
+        	String firstName = "";
         	String displayId = userEId;
         	// if there is last name or first name specified, use it
-        	if (userRoleEntry.lastName != null && userRoleEntry.lastName.length() > 0 
-        			|| userRoleEntry.firstName != null && userRoleEntry.firstName.length() > 0)
-        		userName = userRoleEntry.lastName + "," + userRoleEntry.firstName;
+        	if (userRoleEntry.lastName != null && userRoleEntry.lastName.length() > 0
```
Comment (bjones86) [inline site-manage/site-manage-participant-helper/src/java/org/sakaiproject/site/tool/helper/participant/rsf/DifferentRoleProducer.java]:
```
if( StringUtils.isNotBlank( userRoleEntry.lastName ) && StringUtils.isNotBlank( userRoleEntry.firstName ) )
```

File: site-manage/site-manage-participant-helper/src/java/org/sakaiproject/site/tool/helper/participant/rsf/DifferentRoleProducer.java [hunk 6]
```diff
@@ -127,12 +167,43 @@ public void fillComponents(UIContainer tofill, ViewParameters arg1, ComponentChe
...
 
   		}
-        
-    	UICommand.make(differentRoleForm, "continue", messageLocator.getMessage("gen.continue"), "#{siteAddParticipantHandler.processDifferentRoleContinue}");
+		
+		UISelect.make(differentRoleForm, "role-select-all", roleIds.toStringArray(), handler.sameRoleChoice, String.valueOf(handler.getRoles().get(0)));
+
+		// SAK-26101 only show the status choice when "activeInactiveUser" is set to be true
+		String activeInactiveUser = handler.getServerConfigurationString("activeInactiveUser", Boolean.FALSE.toString());
+		if (activeInactiveUser.equalsIgnoreCase(Boolean.TRUE.toString()))
```
Comment (bjones86) [inline site-manage/site-manage-participant-helper/src/java/org/sakaiproject/site/tool/helper/participant/rsf/DifferentRoleProducer.java]: Flip the operands for null safety

File: site-manage/site-manage-participant-helper/src/java/org/sakaiproject/site/tool/helper/participant/impl/SiteAddParticipantHandler.java (hunk 1 above)
Comment (steveswinsburg) [inline site-manage/site-manage-participant-helper/src/java/org/sakaiproject/site/tool/helper/participant/impl/SiteAddParticipantHandler.java]: Better to use StringUtils or even BooleanUtils

File: site-manage/site-manage-participant-helper/src/java/org/sakaiproject/site/tool/helper/participant/rsf/DifferentRoleProducer.java (hunk 6 above)
Comment (steveswinsburg) [inline site-manage/site-manage-participant-helper/src/java/org/sakaiproject/site/tool/helper/participant/rsf/DifferentRoleProducer.java]: BooleanUtils

**Comment by ern** [comment]
Please provide some screenshots on the JIRA reflecting the UI change.

**Comment by ncaidin** [comment]
Sakai core team review. @davidpbauer can you fix these issues ?  Please attach screenshots and fix. Feel free to reopen. 
Thanks!

**Pull request closed without merging**